harness/determined/common/schemas/expconf/_gen.py

```diff
@@ -2044,8 +2044,14 @@
         "shm_size": {
             "type": [
                 "integer",
+                "string",
                 "null"
             ],
+            "checks": {
+                "must be a valid memory size": {
+                    "pattern": "^([0-9]*[.])?[0-9]+ ?[kmgtpKMGTP]?[iI]?[bB]?$"
```

Comment:
This was their regex which they seemed to migrate away from for performance
https://github.com/docker/go-units/blob/1364ead0d8e38649c7d8d8920d1eafeaf5064071/size.go#L34

I totally agree its wrong because it accepts ```5i``` but at least my```docker run``` at ```20.10.12``` does too. I'll change it to your suggestion since it seems Docker is going to make the (really minor and unimportant) breaking change eventually.

Unrelated but it seems counter intuitive that Docker units treats ```g``` and ```gib``` as the same unit.